src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java

```diff
@@ -64,23 +65,29 @@ public class ArmResponseFilesProcessorImpl implements ArmResponseFilesProcessor
     private final ObjectMapper objectMapper;
     private final UserIdentity userIdentity;
 
-    @Transactional
+    private ObjectRecordStatusEntity armDropZoneStatus;
+    private ObjectRecordStatusEntity armProcessingResponseFilesStatus;
+    private ObjectRecordStatusEntity armResponseProcessingFailed;
+    private ObjectRecordStatusEntity storedStatus;
+    private ObjectRecordStatusEntity checksumFailedStatus;
+    private UserAccountEntity userAccount;
+
     @Override
     public void processResponseFiles() {
+        initialisePreloadedObjects();
         // Fetch All records from external Object Directory table with external_location_type as 'ARM' and status with 'ARM dropzone'.
         ExternalLocationTypeEntity armLocation = externalLocationTypeRepository.getReferenceById(ExternalLocationTypeEnum.ARM.getId());
 
-        ObjectRecordStatusEntity armDropZoneStatus = objectRecordStatusRepository.getReferenceById(ARM_DROP_ZONE.getId());
-        ObjectRecordStatusEntity armProcessingResponseFilesStatus = objectRecordStatusRepository.getReferenceById(ARM_PROCESSING_RESPONSE_FILES.getId());
-
         List<ExternalObjectDirectoryEntity> dataSentToArm =
             externalObjectDirectoryRepository.findByExternalLocationTypeAndObjectStatus(armLocation, armDropZoneStatus);
         if (!CollectionUtils.isEmpty(dataSentToArm)) {
             log.info("ARM Response process found : {} records to be processed", dataSentToArm.size());
             for (ExternalObjectDirectoryEntity externalObjectDirectory : dataSentToArm) {
                 updateExternalObjectDirectoryStatus(externalObjectDirectory, armProcessingResponseFilesStatus);
             }
+            int row = 1;
             for (ExternalObjectDirectoryEntity externalObjectDirectory : dataSentToArm) {
```

Comment:
can we use a more transitional for loop to maintain the count

Reply:
I understand what you are saying but the row number is just being used for logging purposes. If you change it to a for (int row=0; row < dataSentToArm.size(); row++) you always risk the chance you end up skipping an element as the row number can be changed.